### PR TITLE
Add test for seeding algos

### PR DESCRIPTION
**Pull request by bouthilx**

### CODE REVIEW

File: tests/functional/algos/test_algos.py
Comment (lebrice) [inline tests/functional/algos/test_algos.py:156]:
```suggestion
        self, max_trials: int = 30, with_fidelity: bool = False, seed: int | None = None
```

File: tests/functional/algos/test_algos.py
Comment (lebrice) [inline tests/functional/algos/test_algos.py:164]: Oh I see. You're using this method to re-initialize the RNG when reusing the task I think.

Nitpick: I don't remember if the `self.rng` attribute is already type-hinted in the base class. If it isn't, then it's better for the type checker (and also nicer IMO) to declare all attributes in `__init__`.
```suggestion
        self.rng: numpy.random.RandomState
        self.initialize()

    def initialize(self):
        self.rng = numpy.random.RandomState(self.seed)
```

File: tests/functional/algos/test_algos.py
Comment (lebrice) [inline tests/functional/algos/test_algos.py:406]: Is the PR for renaming "algorithms" -> "algorithm" in yet?
```suggestion
        algorithm=algorithm,
```

**Review by lebrice — APPROVED** [review]
Nice!

File: tests/functional/algos/test_algos.py
Comment (bouthilx) [reply tests/functional/algos/test_algos.py:406]: Not yet but good catch. Let's harmonize already. Thanks!

File: tests/functional/algos/test_algos.py
Comment (bouthilx) [inline tests/functional/algos/test_algos.py:410]:
```suggestion
    assert exp.configuration["algorithm"] == algorithm
```

File: tests/functional/algos/test_algos.py
Comment (bouthilx) [inline tests/functional/algos/test_algos.py:422]:
```suggestion
    assert rep_exp.configuration["algorithm"] == algorithm
```

File: tests/functional/algos/test_algos.py
Comment (bouthilx) [inline tests/functional/algos/test_algos.py:418]:
```suggestion
        algorithm=algorithm,
```

**Comment by bouthilx** [comment]
Waiting for #1014 to be merged before merging this PR.

**Pull request merged**
